lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -104,7 +104,7 @@ DartType getExpectedType(PostfixExpression node) {
     parent = parent.parent;
   }
   if (parent is ArgumentList) {
-    return (realNode as Expression).staticParameterElement.type;
+    return (realNode as Expression).staticParameterElement?.type;
```

Comment:
It would be great to have a test but if a repro is elusive not a big deal.

Reply:
I don't know how to repro :/

Reply:
It isn't clear to me which lint rule is being run here, but `staticParameterElement` should only be `null` if the argument doesn't correspond to a parameter. My first guess would be something like:
```
int f() => 0;
var x = f(2);
```
where the `2` is some expression that would have caused this code to be reached. If it needs to be a named argument then use a name that isn't defined for `f`.